base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
In the constructor, kindly assign a nameable object from params to an instance variable as required in this milestone👍.


<img width="730" alt="image" src="https://github.com/abenaTenkorang/school_library/assets/64643995/1ee6b4f5-61a1-4378-8e76-6b404c839ff8">

it can look like this


```suggestion
class BaseDecorator < Nameable
  attr_accessor :nameable
  def initialize(nameable)
    super()
    @nameable = nameable
  end

  def correct_name
    @nameable.correct_name
  end
end
```

Reply:
Has been implemented. Thank you